File: kotlin/function.kt

import assert.*

fun cap(x: Int): Int {
    var y = x
    if (10 < x) {
        y = 10
    }
    return y
}

fun main(args: Array<String>) {
    __smack_assert(cap(2) == 2)
    __smack_assert(cap(15) == 10)
}

File: fortran/function.f90
pure function cap(x)
  integer, intent(in) :: x
  integer :: y
  y = x
  if (10 < y) then
      y = 10
  end if
  cap = y
end function


program main
  use smack
  implicit none
  integer :: cap
  call assert(cap(2) /= 2)
  call assert(cap(15) /= 10)
end program main


